libbpf-tools/memleak.c

```diff
@@ -0,0 +1,707 @@
+		} else {
+			bpf_program__set_autoload(skel->progs.tracepoint_percpu_alloc_percpu, false);
+			bpf_program__set_autoload(skel->progs.tracepoint_percpu_free_percpu, false);
+		}
+	} else {
+		/* Disable all kernel tracepoints*/
+		bpf_program__set_autoload(skel->progs.tracepoint_kmalloc, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_kfree, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_kmalloc_node, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_kmem_cache_alloc, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_kmem_cache_alloc_node, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_kmem_cache_free, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_mm_page_alloc, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_mm_page_free, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_percpu_alloc_percpu, false);
+		bpf_program__set_autoload(skel->progs.tracepoint_percpu_free_percpu, false);
+   }
+	/* Load & verify BPF programs */
+	err = memleak_bpf__load(skel);
+	if (err) {
+		fprintf(stderr, "Failed to load and verify BPF skeleton\n");
+		goto cleanup;
+	}
+
+	/* load kernel symbols */
+	ksyms = ksyms__load();
+	if (!ksyms) {
+		fprintf(stderr, "failed to load kallsyms\n");
+		goto cleanup;
+	}
+	syms_cache = syms_cache__new(0);
+	if (!syms_cache) {
+		fprintf(stderr, "failed to create syms_cache\n");
+		goto cleanup;
+	}
+
+	/* Attach tracepoints */
+	err = memleak_bpf__attach(skel);
+	if (err) {
+		fprintf(stderr, "Failed to attach BPF skeleton\n");
+		goto cleanup;
+	}
+
+	/* Attach tracepoint handler */
+	if (env.kernel_trace) {
+		printf("Attaching to kernel allocators, Ctrl+C to quit.\n");
+	} else {
+		/* find location of libc */
+		if (env.obj != NULL) {
+			if (strlen(env.obj) > PATH_MAX) {
+				fprintf(stderr, "object path too long\n");
+				return -1;
+			}
+			memcpy(libc_path, env.obj, strlen(env.obj));
+		} else {
+			err = get_libc_path(libc_path);
+			if (err) {
+				fprintf(stderr, "could not find libc.so\n");
+				return -1;
+			}
+		}
+
+		printf("Attaching to pid %ld, Ctrl+C to quit.\n", env.pid);
+		INIT_UPROBE_URETPROBE(malloc, env.pid, libc_path, "malloc");
+		INIT_UPROBE_URETPROBE(calloc, env.pid, libc_path, "calloc");
+		INIT_UPROBE_URETPROBE(realloc, env.pid, libc_path, "realloc");
+		INIT_UPROBE_URETPROBE(memalign, env.pid, libc_path, "memalign");
+		INIT_UPROBE_URETPROBE(posix_memalign, env.pid, libc_path, "posix_memalign");
+		INIT_UPROBE_URETPROBE(valloc, env.pid, libc_path, "valloc");
+		INIT_UPROBE_URETPROBE(pvalloc, env.pid, libc_path, "pvalloc");
+		INIT_UPROBE_URETPROBE(aligned_alloc, env.pid, libc_path, "aligned_alloc");
+		INIT_UPROBE(free, env.pid, libc_path, "free");
+	}
+
+	for (i = 0; ; i++) {
+		if (env.count != 0 && i >= env.count) break;
+		sleep(env.interval);
+		if (env.combined_only) {
+	   	print_combined_outstanding_allocations(ksyms, syms_cache, skel);
```

Comment:
Please use tab for indention.

Reply:
Fix in commit [2beb289e](https://github.com/iovisor/bcc/pull/3828/commits/2beb289e98d651bddc2127299c5627b5d6149fce).